main-settings/src/main/scala/sbt/std/TaskLinterDSL.scala

```diff
@@ -57,7 +58,7 @@ abstract class BaseTaskLinterDSL extends LinterDSL {
       }
 
       @inline def isKey(tpe: Type): Boolean =
-        tpe <:< taskKeyType || tpe <:< settingKeyType || tpe <:< inputKeyType
+        tpe <:< initializeType || tpe <:< taskKeyType || tpe <:< settingKeyType || tpe <:< inputKeyType
```

Comment:
this should simplify to just `tpe <:< initializeType` because keys are initializes

Reply:
Good point. Let me try that.